### PR TITLE
[TEVA-2674] Show dashboard link when there are draft applications

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-2674

### Before
Screenshot 2021-05-25 at 11 40 33

### After
Screenshot 2021-05-25 at 11 40 08

### CODE REVIEW

File: app/components/publishers/vacancies_component.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def view_applicants(vacancy, card)
                            organisation_job_job_applications_path(vacancy.id),
                            class: "govuk-link--no-visited-state")
       tag.div(card.labelled_item(I18n.t("jobs.manage.applications"), link))
-    elsif vacancy.job_applications.withdrawn.any? || vacancy.job_applications.none?
+    elsif vacancy.job_applications.withdrawn.any? || vacancy.job_applications.draft.any? || vacancy.job_applications.none?
```
Comment (cpjmcquillan) [inline app/components/publishers/vacancies_component.rb:56]: It's only the count that's different isn't it? Could we rewrite the method to make it a bit simpler?

Comment (david-mears-dfe) [reply app/components/publishers/vacancies_component.rb]: I'm confused by it - why isn't it equivalent to `else`?

Comment (cesidio) [reply app/components/publishers/vacancies_component.rb]: code is simplified now

File: app/components/publishers/vacancies_component.rb [hunk 2]
```diff
@@ -47,18 +47,12 @@ def view_applicants(vacancy, card)
     return unless include_job_applications?
 
     applications = vacancy.job_applications.where(status: %w[submitted reviewed shortlisted unsuccessful])
```
Comment (david-mears-dfe) [inline app/components/publishers/vacancies_component.rb:49]: Could you do the `.count` on this line to save space?

**Review by cpjmcquillan — APPROVED** [review]

**Pull request merged**